examples/scripts/superres_ppp_dncnn_admm.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# This file is part of the SCICO package. Details of the copyright
+# and user license can be found in the 'LICENSE.txt' file distributed
+# with the package.
+
+"""
+Image Superresolution (ADMM Plug-and-Play Priors w/ DnCNN)
+==========================================================
+
+This example demonstrates the use of the ADMM Plug and Play Priors
+(PPP) algorithm :cite:`venkatakrishnan-2013-plugandplay2` for solving
+a simple image superresolution problem.
+"""
+
+import jax
+
+import scico
+import scico.numpy as snp
+import scico.random
+from scico import denoiser, functional, linop, loss, metric, plot
+from scico.data import kodim23
+from scico.optimize.admm import ADMM, LinearSubproblemSolver
+from scico.solver import cg
+from scico.util import device_info
+
+"""
+Define downsampling function.
+"""
+
+
+def downsample_image(img, rate):
+    img = snp.mean(snp.reshape(img, (-1, rate, img.shape[1], img.shape[2])), axis=1)
+    img = snp.mean(snp.reshape(img, (img.shape[0], -1, rate, img.shape[2])), axis=2)
+    return img
+
+
+"""
+Read a ground truth image.
+"""
+img = kodim23(asfloat=True)[160:416, 60:316]
+img = jax.device_put(img)
+
+
+"""
+Create a test image by downsampling and adding Gaussian white noise.
+"""
+rate = 4  # downsampling rate
+σ = 2e-2  # noise standard deviation
+
+Afn = lambda x: downsample_image(x, rate=rate)
+s = Afn(img)
+inshape = img.shape
```

Comment:
May as well call then `input_shape` and `output_shape`?

Reply:
Fair point; will do.